tests/test_p4_submit.py

```diff
@@ -37,3 +38,23 @@ def test_success_changing_checked_out_file(p4_submit_environment):
     p4_submit_environment.p4.run_submit(change)
 
     assert p4_submit_environment.file_present(str(target_file))
+
+
+@pytest.mark.parametrize("test_type,expected",
+                         [("open", 0), ("write", 1), ("review", 1)],
+                         ids=["open", "write", "review"])
+def test_fail_protected_branch(p4_submit_environment, test_type, expected):
+    protected_dir = p4_submit_environment.vcs_cooking_dir.mkdir(test_type + "-protected")
+    file_to_add = protected_dir.join("new_file.txt")
+    text = "This is a new line in the file"
+    file_to_add.write(text + "\n")
+
+    settings = copy.deepcopy(p4_submit_environment.settings)
+    setattr(settings.Submit, "reconcile_list", [unicode(file_to_add)])
+
+    result = universum.run(settings)
+    assert result == expected
+
+    p4 = p4_submit_environment.p4
+    assert not p4.run_changes("-c", p4_submit_environment.client_name, "-s", "pending")
```

Comment:
Please document what these asserts check exactly.
Otherwise, the reader has to google for p4 documentation.